pkg/apis/serving/v1/revision_defaults_test.go

```diff
@@ -1063,6 +1063,251 @@ func TestRevisionDefaulting(t *testing.T) {
+									Command: []string{"echo", "hi"},
+								},
+							},
+						},
+					}, {
+						Name:      config.DefaultUserContainerName + "-1",
+						Resources: defaultResources,
+						ReadinessProbe: &corev1.Probe{
+							SuccessThreshold: 1,
+							ProbeHandler: corev1.ProbeHandler{
+								Exec: &corev1.ExecAction{
+									Command: []string{"echo", "hi"},
+								},
+							},
+						},
+					}},
+				},
+			},
+		},
+	}, {
+		name: "multiple containers apply k8s defaults when period seconds has a non zero value",
+		in: &Revision{
+			Spec: RevisionSpec{
+				PodSpec: corev1.PodSpec{
+					Containers: []corev1.Container{{
+						Ports: []corev1.ContainerPort{{
+							ContainerPort: 8080,
+						}},
+						ReadinessProbe: &corev1.Probe{
+							// FailureThreshold and TimeoutSeconds missing
+							PeriodSeconds: 10,
+						},
+					}, {
+						ReadinessProbe: &corev1.Probe{
+							// FailureThreshold and TimeoutSeconds missing
+							PeriodSeconds: 10,
+						},
+					}},
+				},
+			},
+		},
+		want: &Revision{
+			Spec: RevisionSpec{
+				ContainerConcurrency: ptr.Int64(config.DefaultContainerConcurrency),
+				PodSpec: corev1.PodSpec{
+					Containers: []corev1.Container{{
+						Name: config.DefaultUserContainerName + "-0",
+						Ports: []corev1.ContainerPort{{
+							ContainerPort: 8080,
+						}},
+						ReadinessProbe: &corev1.Probe{
+							FailureThreshold: 3, // Added as k8s default
+							ProbeHandler:     defaultProbe.ProbeHandler,
+							PeriodSeconds:    10,
+							SuccessThreshold: 1,
+							TimeoutSeconds:   1, // Added as k8s default
+						},
+						Resources: defaultResources,
+					}, {
+						Name: config.DefaultUserContainerName + "-1",
+						ReadinessProbe: &corev1.Probe{
+							FailureThreshold: 3, // Added as k8s default
+							PeriodSeconds:    10,
+							SuccessThreshold: 1,
+							TimeoutSeconds:   1, // Added as k8s default
+						},
+						Resources: defaultResources,
+					}},
+				},
+				TimeoutSeconds: ptr.Int64(config.DefaultRevisionTimeoutSeconds),
+			},
+		},
+	}, {
+		name: "multiple containers partially initialized",
+		in: &Revision{
+			Spec: RevisionSpec{
+				PodSpec: corev1.PodSpec{Containers: []corev1.Container{{
+					Ports: []corev1.ContainerPort{{
+						ContainerPort: 8080,
+					}},
+				}, {}}},
+			},
+		},
+		want: &Revision{
+			Spec: RevisionSpec{
+				TimeoutSeconds:       ptr.Int64(config.DefaultRevisionTimeoutSeconds),
+				ContainerConcurrency: ptr.Int64(config.DefaultContainerConcurrency),
+				PodSpec: corev1.PodSpec{
+					Containers: []corev1.Container{{
+						Name: config.DefaultUserContainerName + "-0",
+						Ports: []corev1.ContainerPort{{
+							ContainerPort: 8080,
+						}},
+						Resources:      defaultResources,
+						ReadinessProbe: defaultProbe,
+					}, {
+						Name:           config.DefaultUserContainerName + "-1",
+						Resources:      defaultResources,
+						ReadinessProbe: nil,
+					}},
+				},
+			},
+		},
```

Comment:
Could we add another test case to compare against liveness probes for which we dont do any defaulting:

```
{
		name: "multiple containers with probes and overrides",
		in: &Revision{
			Spec: RevisionSpec{
				PodSpec: corev1.PodSpec{
					Containers: []corev1.Container{{
						ReadinessProbe: &corev1.Probe{
							ProbeHandler: corev1.ProbeHandler{
								Exec: &corev1.ExecAction{
									Command: []string{"echo", "hi"},
								},
							},
						},
					}, {
						LivenessProbe: &corev1.Probe{
							PeriodSeconds: 10,
							ProbeHandler: corev1.ProbeHandler{
								//Exec: &corev1.ExecAction{
								//	Command: []string{"echo", "hi"},
								//},
								TCPSocket: &corev1.TCPSocketAction{
									Host: "127.0.0.2",
								},
							},
						},
					}},
				},
			},
		},
		want: &Revision{
			Spec: RevisionSpec{
				TimeoutSeconds:       ptr.Int64(config.DefaultRevisionTimeoutSeconds),
				ContainerConcurrency: ptr.Int64(config.DefaultContainerConcurrency),
				PodSpec: corev1.PodSpec{
					Containers: []corev1.Container{{
						Name:      config.DefaultUserContainerName + "-0",
						Resources: defaultResources,
						ReadinessProbe: &corev1.Probe{
							SuccessThreshold: 1,
							ProbeHandler: corev1.ProbeHandler{
								Exec: &corev1.ExecAction{
									Command: []string{"echo", "hi"},
								},
							},
						},
					}, {
						Name:      config.DefaultUserContainerName + "-1",
						Resources: defaultResources,
						LivenessProbe: &corev1.Probe{
							SuccessThreshold: 0, // Not touched only for readiness probes
							PeriodSeconds:    10,
							ProbeHandler: corev1.ProbeHandler{
								TCPSocket: &corev1.TCPSocketAction{
									Host: "127.0.0.2",
								},
							},
						},
					}},
				},
			},
		},
	}```

Reply:
I'd also argue to tackle this in a separate issue/PR. I'd rather not change existing behaviour in this PR.